internal/controllers/topology/cluster/desired_state.go

```diff
@@ -424,10 +424,14 @@ func (r *Reconciler) computeControlPlaneVersion(ctx context.Context, s *scope.Sc
 	}
 
 	// If the control plane is not upgrading or scaling, we can assume the control plane is stable.
-	// However, we should also check for the MachineDeployments to be stable.
-	// If the MachineDeployments are rolling out (still completing a previous upgrade), then do not pick
-	// up the desiredVersion yet. We will pick up the new version after the MachineDeployments are stable.
-	if s.Current.MachineDeployments.IsAnyRollingOut() {
+	// However, we should also check for the MachineDeployments upgrading.
+	// If the MachineDeployments are upgrading, then do not pick up the desiredVersion yet.
+	// We will pick up the new version after the MachineDeployments finish upgrading.
+	mdUpgradingNames, err := s.Current.MachineDeployments.Upgrading(ctx, r.Client)
```

Comment:
Q: Can we run

```
	upgradingMDs, err := s.Current.MachineDeployments.Upgrading(ctx, r.Client)
	if err != nil {
		return nil, err
	}
	s.UpgradeTracker.MachineDeployments.MarkUpgradingAndRollingOut(upgradingMDs...)
```

Once early in the compute and then just use the results in computeMDVersion / computeCPVersion?